src/algebra/invertible.lean

```diff
@@ -104,15 +104,22 @@ def unit_of_invertible [monoid α] (a : α) [invertible a] : units α :=
 lemma is_unit_of_invertible [monoid α] (a : α) [invertible a] : is_unit a :=
 ⟨unit_of_invertible a, rfl⟩
 
+/-- Units are invertible in their associated monoid. -/
+instance units.invertible [monoid α] (u : units α) : invertible (u : α) :=
+{ inv_of := ↑(u⁻¹), inv_of_mul_self := u.inv_mul, mul_inv_of_self := u.mul_inv }
+
+@[simp] lemma inv_of_units [monoid α] (u : units α) : ⅟(u : α) = ↑(u⁻¹) :=
+inv_of_eq_right_inv u.mul_inv
+
 /-- Each element of a group is invertible. -/
-def invertible_of_group [group α] (a : α) : invertible a :=
+instance invertible_of_group [group α] (a : α) : invertible a :=
```

Comment:
Way back in [the Zulip discussion](https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/.232480.20bilinear.20forms) about #2504, @sgouezel [noted that](https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/.232480.20bilinear.20forms/near/195066068) `invertible_of_group` and `invertible_one` give two non-defeq values of `invertible 1`, so they shouldn't both be instances.

I'm not quite sure how to resolve this, to be honest: there's definitely value in keeping `invertible` around as the computable alternative to `is_unit`, but perhaps using it as a typeclass just isn't going to work because of coherence issues.